Analyser/src/bsa_analyser/github/io/FileValidator.java

```diff
@@ -5,24 +5,79 @@
  */
 package bsa_analyser.github.io;
 
-import java.io.BufferedReader;
-import java.io.FileReader;
+import java.io.*;
+
 import static java.lang.Boolean.FALSE;
 import static java.lang.Boolean.TRUE;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
 import javax.swing.JOptionPane;
 
 /**
  *
- * @author jkell
+ * @author jkelly, Valetina, and Tolu
  */
 
 
 public class FileValidator {
-    
-    
+    public static void fasta_content_checker(File user_file) {
+
+    boolean b = false;
+
+    List<String> Fasta_Content = new ArrayList<String>();
+        if (user_file.getName().endsWith("fasta") || user_file.getName().endsWith("fa")) {
+            String line = "";
+            try {
+
+                BufferedReader br = new BufferedReader(new FileReader((user_file)));
+                while (((line = br.readLine()) != null)) {
+
+                    if ((!line.contains("A|T|G|C|N")) && (!line.startsWith(">"))) {
+                        //while (((line = br.readLine() != null)) {
+                        for (int i = 0; i < line.length(); i++) {
+                            if (!(line.substring(i, i + 1).matches(("|A|T|G|C|N|a|t|g|c|n")))) {
+                                int user_option = JOptionPane.showConfirmDialog(null, "This Fasta file is corrupt"
+                                        + " This means the content may not be a Fasta file format. Do you wish to upload another file?"
+                                        + " (error code 1)");
+
+                                if (user_option == 0) {
```

Comment:
Having this logic in  a while loop means that if the user doesn't want to upload the file it keeps asking if they want to upload another file - it is impossible to break out of the loop!